tensorflow/compiler/xla/service/gpu/gemm_rewriter.cc

```diff
@@ -373,6 +362,31 @@ class GemmRewriterVisitor : public DfsHloRewriteVisitor {
     return OkStatus();
   }
 
+  Status ExchangeBinaryOpAndSlice(HloInstruction *instr, HloInstruction *slice,
```

Comment:
Can we RET_CHECK that the relevant instructions here have only one use?  (I don't think it's *all* of the instructions that need to have one use.  For example, if the broadcast has multiple uses that should be fine.)

This is to enforce an invariant that you should be checking elsewhere.  My intent is to make it easier for readers to understand the invariants of this specific function.